### PR TITLE
[AU4] restruct of playback

### PR DESCRIPTION
**Pull request by igorkorsukov**

### CODE REVIEW

**Comment by igorkorsukov** [comment]
> Explain please?

Synchronized with the structure in MuseScore.
Our goal is to make applications similar with MuseScore, so as not to waste time on what has already been implemented and to be able to focus on the unique functions of Audacity.

**Comment by Paul-Licameli** [comment]
> > Explain please?
> 
> Synchronized with the structure in MuseScore. Our goal is to make applications similar with MuseScore, so as not to waste time on what has already been implemented and to be able to focus on the unique functions of Audacity.

How deep is this work?  Will it rewrite the inter-thread communications that happen now in AudioIO.cpp?

**Comment by igorkorsukov** [comment]
> How deep is this work? Will it rewrite the inter-thread communications that happen now in AudioIO.cpp?

Not now, at the moment these are only wrappers for AU3 (AudioIO)

File: au4/src/playback/internal/playbackcontroller.cpp [hunk 1]
```diff
@@ -179,78 +175,62 @@ void PlaybackController::togglePlay()
...
-    msecs_t endMsecs = playbackEndMsecs();
-    msecs_t newPosition = !args.empty() ? args.arg<msecs_t>(0) : 0;
-    newPosition = std::clamp(newPosition, startMsecs, endMsecs);
+    // msecs_t startMsecs = playbackStartMsecs();
+    // msecs_t endMsecs = playbackEndMsecs();
+    // msecs_t newPosition = !args.empty() ? args.arg<msecs_t>(0) : 0;
+    // newPosition = std::clamp(newPosition, startMsecs, endMsecs);
 
-    seek(newPosition);
+    // seek(newPosition);
```
Comment (Eism) [inline au4/src/playback/internal/playbackcontroller.cpp:201]: can you please restore this?

**Comment by igorkorsukov** [comment]
> “Not now” suggests that you mean “later.”
> 
> Are you sure you won’t discover things AudioIO.cpp does that MuseScore does not and have to rethink this?

We have not yet explored this issue, but we plan to study it in the future, and then we will see.

File: au4/src/playback/internal/playbackcontroller.cpp (hunk 1 above)
Comment (igorkorsukov) [reply au4/src/playback/internal/playbackcontroller.cpp:201]: The method is called rewindToStart, but we accept an optional position, should we change the name of the method or remove the optional position? In what cases can a position be put?

Comment (Eism) [reply au4/src/playback/internal/playbackcontroller.cpp:201]: let's remove the optional position

Comment (igorkorsukov) [reply au4/src/playback/internal/playbackcontroller.cpp]: fixed

**Review by Eism — APPROVED** [review]

**Pull request merged**